go/vt/mysqlctl/builtinbackupengine.go

```diff
@@ -512,18 +512,23 @@ func (be *BuiltinBackupEngine) backupFiles(
 		wg.Add(1)
 		go func(i int) {
 			defer wg.Done()
-
-			// Wait until we are ready to go, skip if we already
-			// encountered an error.
-			sema.Acquire(ctx, 1)
+			fe := &fes[i]
+			// Wait until we are ready to go, return if we encounter an error
+			acqErr := sema.Acquire(ctx, 1)
+			if acqErr != nil {
+				log.Errorf("not able to acquire semaphore to backup file:%s  err: %s", fe.Name, acqErr.Error())
+				bh.RecordError(acqErr)
+				return
+			}
```

Comment:
https://pkg.go.dev/golang.org/x/sync/semaphore#Weighted.Acquire 
> Acquire acquires the semaphore with a weight of n, blocking until resources are available or ctx is done. On success, returns nil. On failure, returns ctx.Err() and leaves the semaphore unchanged.
>
> If ctx is already done, Acquire may still succeed without blocking.

So I think we also need to check for context cancellation here. Something like this:

```
	acqErr := sema.Acquire(ctx, 1)
	if acqErr != nil {
		log.Errorf("Not able to acquire semaphore to backup file:%s  err: %s", fe.Name, acqErr.Error())
		bh.RecordError(acqErr)
		return
	}
	select {
	case <-ctx.Done():
		log.Errorf("Context cancelled during %q backup", bh.Name())
		bh.RecordError(vterrors.Errorf(vtrpcpb.Code_CANCELED, "context has expired"))
		return
	default:
	}
```

Reply:
Yes, I found out this was causing flakiness in the test "TestExecuteBackupWithCancelledContext". So reached to same conclusion :)

Reply:
done